src/Nethermind/Nethermind.Facade/Eth/RpcTransaction/TransactionForRpc.cs

```diff
@@ -76,77 +83,95 @@ public JsonConverter()
             RegisterTransactionType<SetCodeTransactionForRpc>();
         }
 
-        internal static void RegisterTransactionType<T>() where T : TransactionForRpc, ITxTyped
-            => _types[(int)T.TxType] = typeof(T);
+        internal static void RegisterTransactionType<T>() where T : TransactionForRpc, IFromTransaction<T>, ITxTyped
+        {
+            Type txType = typeof(T);
+            string[] uniqueProperties = txType.GetProperties(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
+                .Where(p => p.GetCustomAttribute<JsonDiscriminatorAttribute>() is not null)
+                .Select(p => p.Name).ToArray();
+
+            TxTypeInfo typeInfo = new TxTypeInfo
+            {
+                TxType = T.TxType,
+                Type = txType,
+                FromTransactionFunc = T.FromTransaction,
+                DiscriminatorProperties = uniqueProperties
+            };
+
+            int existingTypeInfo = _txTypes.FindIndex(t => t.Type == txType);
+
+            if (existingTypeInfo != -1)
+            {
+                // Adding in reverse order so new tx type are in priority
+                _txTypes[existingTypeInfo] = typeInfo;
+            }
+            else
+            {
+                _txTypes.Insert(0, typeInfo);
+            }
```

Comment:
I think we shouldn't depend on addition order, but tx type order. Then we need that list kept sorted. Either sort it after adding or use linked list and inserting while keeping it sorted.

Reply:
> Properties marked with <see cref="JsonDiscriminatorAttribute" /> are checked in reverse order of tx type registration.

Looks like it's not enough to mention this, and others will assume different behavior. No problem I will change it.